cmd/xk6-depsync/main.go

```diff
@@ -0,0 +1,107 @@
+// main is the main package for xk6-depsync, a script that checks and provides commands to synchronize
+// common dependencies with k6 core.
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"sort"
+	"strings"
+
+	"golang.org/x/mod/modfile"
+)
+
+const (
+	k6Core     = "go.k6.io/k6"
+	k6GoModURL = "https://proxy.golang.org/" + k6Core + "/@v/%s.mod"
+)
+
+func main() {
+	gomod := "go.mod"
+	if len(os.Args) >= 2 {
+		gomod = os.Args[1]
+	}
+
+	file, err := os.Open(gomod)
+	if err != nil {
+		log.Fatalf("opening local go.mod: %v", err)
+	}
+
+	ownDeps, err := dependencies(file)
+	if err != nil {
+		log.Fatalf("reading dependencies: %v", err)
+	}
+
+	k6Version := ownDeps[k6Core]
+	if k6Version == "" {
+		log.Fatalf("k6 core %q not found in %q", k6Core, gomod)
+	}
+
+	log.Printf("detected k6 core version %s", k6Version)
+
+	k6CoreVersionedURL := fmt.Sprintf(k6GoModURL, k6Version)
+	//nolint:bodyclose // Single-run script.
+	response, err := http.Get(k6CoreVersionedURL)
+	if err != nil {
+		log.Fatalf("error fetching k6 go.mod: %v", err)
+	}
+
```

Comment:
```suggestion
	response, err := http.Get(k6CoreVersionedURL)
	if err != nil {
		log.Fatalf("error fetching k6 go.mod: %v", err)
	}
	defer func() { _ = response.Body.Close() }()
```
Given that it is still one line - closing it seems like the better approach IMO.